server/cache.go

```diff
@@ -32,6 +32,48 @@ var (
 	}
 )
 
+type storesInfo struct {
+	stores map[uint64]*storeInfo
+}
+
+func newStoresInfo() *storesInfo {
+	return &storesInfo{
+		stores: make(map[uint64]*storeInfo),
+	}
+}
+
+func (s *storesInfo) getStore(storeID uint64) *storeInfo {
+	store, ok := s.stores[storeID]
+	if !ok {
+		return nil
+	}
+	return store.clone()
```

Comment:
The mutex only protects the `storesInfo`, it will not protect the store we returns. If we returns the original store, the caller can still modify the store whenever it wants to.